### DISCUSSION TITLE
Usability questions and tips

### DISCUSSION THREAD

**Discussion by Nick1296** [vlaci/git-agecrypt#3] [Q&A]
Hi!

First of all thanks for the amazing tool! I am using it in tandem with sops-nix to avoid spilling secrets in my NixOS 23.11 config! (I prefer not to have IPs of my self-hosted services in plain sight and I can't use sops to encrypt the nix files that contain those IPs).

 I installed git-agecrypt from the flake you provided and I noticed a couple of things:
-  There is a typo in the implementation of `git -agecrypt deinit`  the section to remove is `filter.git-agecrypt` but the program instead  searches for `fiter.git-agecrypt` and goes in error.
- `git config add -i` is designed to take in paths of private keys but in step 4 of the usage section of the readme a public key is given (this is just a readme fix).
- When configuring the repo for the first time it was impossible to add a new file and have it encrypted directly. I had to resort to:
  1.  Make sure `.gitattributes` has no matching git-agecrypt filter for the file
  2.  add and commit the file without the content I wanted encrypted (an empty file would be good too I guess)
  3.  add the git-agecrypt filters in `.gitattributes` to match the file I just added.
  4. put the content I needed to be encrypted and commit again
  5.  It works!
 
It would be nice to have these things more streamlined when you have the time.

Thanks again!

**Comment by GTrunSec** [comment]
@vlaci What do you think about these ideas? I would like to add `git-crypt` to the upstream of nixpkgs.

**Reply by fidgetingbits** [reply]
Definitely clearer instructions/examples would be useful. I've yet to have it encrypt anything, including following @nick1296 steps above.

**Comment by jordan-bravo** [comment]
@Nick1296 , when I tried your approach, when I get to step iv (I add content that I want to be encrypted to the previously committed blank file), it shows this error:
```
Error: No public key can be found for '/home/main/.nix/secrets/git-agecrypt-secrets.nix'
error: external filter '/nix/store/4xda88asn6zn5f9id80gslrpfj423gr8-git-agecrypt-unstable-2024-03-11/bin/git-agecrypt clean -f %f' failed 1
error: external filter '/nix/store/4xda88asn6zn5f9id80gslrpfj423gr8-git-agecrypt-unstable-2024-03-11/bin/git-agecrypt clean -f %f' failed
fatal: secrets/git-agecrypt-secrets.nix: clean filter 'git-agecrypt' failed
```

Does anyone have any suggestions on what the problem might be?

**Reply by Titaniumtown** [reply]
I've been getting constant issues with this tool, including what you're describing :(

**Reply by Nick1296** [reply]
Have you used both git agecript add -i and git agecrypt config add -r ?
Also you need to have the file in your git tree already before setting up the .gitattributes

**Reply by Titaniumtown** [reply]
Yup, I've tried all of that. I ended up moving to git-crypt and that works fine for my purposes.

**Comment by Nick1296** [comment]
Understood thanks for the info! I will look into that if I start to have
problems too
Mattia Nicolella
